src/main/java/com/gildedrose/Item.java

```diff
@@ -14,6 +14,18 @@ public Item (String name, int sellIn, int quality) {
         this.quality = quality;
     }
 
+    public void updateQuality() {
```

Comment:
Something I should have spotted before - it's useful to add `@Override` to the methods in the subclasses which override this method. It has a few advantages: it makes the relationship between classes clearer, and it adds a bit of compile-time safety because the compiler will complain if you try to delete a method which subclasses override.

Ideally, Java would _require_ you to add this annotation, but for backwards-compatibility reasons it doesn't.

See item 40 of the Effective Java book for more on this.